test/api/input_file_test.cpp

```diff
@@ -190,7 +192,55 @@ TEST(input_file, long_read_sam_file_unsorted)
                        default_min_length,
                        default_max_overlap};
     EXPECT_THROW(detect_junctions_in_long_reads_sam_file(junctions_res,
+                                                         references_lengths,
                                                          args), seqan3::format_error);
 
     std::filesystem::remove(unsorted_sam_path);
 }
+
+TEST(input_file, short_and_long_read_sam_file_with_different_references_lenghts)
+{
+    testing::internal::CaptureStdout();
+    testing::internal::CaptureStderr();
+
+    std::string const expected_err
+    {
+        "Warning: The reference id testchr was found twice in the input files with different length: 1000 and 1001\n"
+        "The cigar string method for short reads is not yet implemented.\n"
+        "The split read method for short reads is not yet implemented.\n"
+        "The read depth method for short reads is not yet implemented.\n"
+        "Warning: The reference id testchr was found twice in the input files with different length: 1000 and 1001\n"
+        "The read depth method for long reads is not yet implemented.\n"
+    };
+
+    std::vector<Junction> junctions_res{};
+    std::map<std::string, int32_t> references_lengths{};
+
+    // Create a blank SAM file with SQ header tag with different length of one reference.
+    std::filesystem::path const tmp_dir = std::filesystem::temp_directory_path();     // get the temp directory
+    std::filesystem::path sam_path{tmp_dir/"unsorted.sam"};
+    std::ofstream sam{sam_path.c_str()};
+    sam << "@HD\tVN:1.6\tSO:coordinate\n" <<
+           "@SQ\tSN:testchr\tLN:1000\n" <<
+           "@SQ\tSN:testchr\tLN:1001\n" <<
+           "test1\t16\ttestchr\t1\t60\t10M\t=\t1\t0\tGCGCGCGCGC\tFFFFFFFFFF\n";
```

Comment:
Nice test case 👍 
However, it is more likely that we have two different SAM files that disagree concerning the length of testchr. It should be easy to create two SAMs instead of one and plug them into `detect_junctions_in_short_reads_sam_file` and `detect_junctions_in_long_reads_sam_file`, respectively.

Reply:
I have extended the testcase a bit to cover more edgecases.